TAO/TAO_IDL/ast/ast_expression.cpp

```diff
@@ -1736,7 +1742,8 @@ AST_Expression::eval_bin_op (AST_Expression::EvalKind ek)
       return nullptr;
     }
 
-  ExprType const expr_type = eval_kind_to_expr_type (ek);
+  ExprType const expr_type = ek == EK_const ?
+    pd_v1->ev()->et : eval_kind_to_expr_type (ek);
```

Comment:
```suggestion
    pd_v1->ev ()->et : eval_kind_to_expr_type (ek);
```
at least 1 more of these below